author	Mikel Astiz <mikel.astiz@bmw-carit.de>
commit	4523901db271776f1e5972dea6e982903ed4db5c
	Mon, 7 May 2012 14:05:39 +0000 (07:05 -0700)

audio: Fix gateway state check

Gateway should be considered active also if connecting or playing.

This could for example lead to manager_find_device() not returning a
device that is connecting, and thus the corresponding endpoint would
never be created in the Media API.

diff -cr bluez-4.99.pan.orig/audio/device.c bluez-4.99.pan/audio/device.c
*** bluez-4.99.pan.orig/audio/device.c	Wed Dec 21 17:53:54 2011
--- bluez-4.99.pan/audio/device.c	Mon Oct 15 05:46:07 2012
***************
*** 701,707 ****
  				control_is_active(dev))
  		return TRUE;
  	else if (!strcmp(interface, AUDIO_GATEWAY_INTERFACE) && dev->gateway &&
! 				gateway_is_connected(dev))
  		return TRUE;
  
  	return FALSE;
--- 701,707 ----
  				control_is_active(dev))
  		return TRUE;
  	else if (!strcmp(interface, AUDIO_GATEWAY_INTERFACE) && dev->gateway &&
! 				gateway_is_active(dev))
  		return TRUE;
  
  	return FALSE;
